pages/api/user/[id].ts

```diff
@@ -25,8 +26,28 @@ export default async function handle(
         }
         ResponseUtil.returnOK(res, user);
         return;
+    } else if (req.method == "PUT") {
+        const session = await getSession({ req });
+        const userId = session ? session.id : req.query.id;
+        const updatedUserData = {
+            id: userId,
+            first_name: req.body.first_name,
+            last_name: req.body.last_name,
+            role: req.body.role,
```

Comment:
What's `role` vs `role_data`?

Reply:
`role` is the enum name of the role (i.e., `"PARENT"`, `"VOLUNTEER"`, etc.) whereas the `role_data` is the actual data corresponding to that role (e.g., address, phone number, etc.)